src/yardstick/artifact.py

```diff
@@ -459,14 +489,14 @@ def __hash__(self) -> int:
                 self.note,
                 # note: we cannot depend on the dataclasses generated hash since fullentry_fields must be hashable
                 tuple(sorted(self.fullentry_fields)),
-            )
+            ),
         )
 
     def __eq__(self, other: Any) -> bool:
         return hash(self) == hash(other)
 
-    def matches_image(self, image: str):
-        if not self.image:
+    def matches_image(self, image: Optional[str]):
+        if not self.image or not image:
```

Comment:
correct, but was directly due to a linter error -- mypy discovered there was an issue since `image` was not `Optional`, which highlighted this extra case, where an unspecified image should always match.